realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -451,28 +556,43 @@ protected final boolean checkAndReportInvalidState() {
             // TODO: Should we expose the underlying Realm errors directly? What else would make sense?
             boolean managementErrorHappened;
             boolean permissionErrorHappened;
+            boolean defaultPermissionErrorHappened;
             ObjectServerError managementError;
             ObjectServerError permissionError;
+            ObjectServerError defaultPermissionError;
             synchronized (permissionManager.errorLock) {
                 // Only hold lock while making a safe copy of current error state
                 managementErrorHappened = (permissionManager.managementRealmError != null);
                 permissionErrorHappened = (permissionManager.permissionRealmError != null);
+                defaultPermissionErrorHappened = (permissionManager.defaultPermissionsRealmError != null);
                 managementError = permissionManager.managementRealmError;
                 permissionError = permissionManager.permissionRealmError;
+                defaultPermissionError = permissionManager.defaultPermissionsRealmError;
             }
 
```

Comment:
I think it's better to have following early return here to avoid `LinkedHashMap` allocation on no errors.

```
if (!permissionErrorHappened && !managementErrorHappened && !defaultPermissionErrorHappened) {
    // Everything seems valid
    return false;
}
```